File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        int cur=0;
        return sumNode(root,cur);
    }
    int sumNode(TreeNode* root,int cur){
        if (root==NULL) return 0;
        cur=cur*10+root->val;
        if (root->left==NULL && root->right==NULL) return cur;
        int left=sumNode(root->left,cur);
        int right=sumNode(root->right,cur);
        return left+right;
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        
        
        def sumNode(root, cur):
            if root == None: return 0
            cur = cur*10+root.val
            if root.left == None and root.right == None: return cur
            left = sumNode(root.left, cur)
            right = sumNode(root.right, cur)
            return left+right
        
        cur = 0
        return sumNode(root, cur)